src/pyk/kore/rpc.py

```diff
@@ -423,10 +424,16 @@ def __init__(self, error: str, context: Iterable[str]):
 @dataclass
 class SmtSolverError(KoreClientError):
     pattern: Pattern
+    definition_dir: Path | None
 
-    def __init__(self, pattern: Pattern):
+    def __init__(self, pattern: Pattern, definition_dir: Path | None = None):
         self.pattern = pattern
-        super().__init__(f'Smt solver error: {self.pattern.text}')
+        self.definition_dir = definition_dir
+        if self.definition_dir is not None:
+            printer = KPrint(self.definition_dir)
+            super().__init__(f'Smt solver error: {printer.kore_to_pretty(self.pattern)}')
```

Comment:
I'd recommend making "SMT" uppercase since it's an abbreviation, here and below:
```suggestion
            super().__init__(f'SMT solver error: {printer.kore_to_pretty(self.pattern)}')
```